.github/workflows/prepare-new-release.yml

```diff
@@ -0,0 +1,149 @@
+        type: string
+        default: main
+      base_ref_version:
+        description: 'verify current dev version (w/o "-dev" suffix)'
+        required: true
+        type: string
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+      RELEASE_VERSION: ${{ github.event.inputs.version_to_release }}
+      NEXT_DEV_VERSION: "${{ github.event.inputs.next_version }}-dev"
+      TAG_NAME: "v${{ github.event.inputs.version_to_release }}"
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+    - name: Configure git
+      run: |
+        set -euo pipefail
+        git config --global user.name 'github-actions[bot]'
+        git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.inputs.base_ref }}
+
+    - name: Check that tag is available
+      run: |
+        if gh release view "$TAG_NAME" ; then
+          echo "::error ::tag $TAG_NAME already exists"
+          exit 1
+        fi
+
+    - name: Check Cargo.toml version
+      run: |
+        set -euo pipefail
+        toml_current_version=$(grep '^version' Cargo.toml | sed 's/version = "\(.*\)"/\1/')
+        expect_version="${VERIFY_VERSION}-dev"
+        if [[ "$toml_current_version" != "$expect_version" ]]; then
+          echo "::error title=bad version::Expected version $expect_version does not match current version $toml_current_version."
+          exit 1
+        fi
+      env:
+        VERIFY_VERSION: ${{ github.event.inputs.base_ref_version }}
+
+    - name: Set up Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        override: true
+
+    - name: Release Cargo.toml
+      run: |
+        sed -i 's/^version = ".*"/version = "${{ env.RELEASE_VERSION }}"/' Cargo.toml
+        cargo metadata >/dev/null
+
+    - name: Release Commit
+      run: |
+        set -euo pipefail
+        git checkout -b "release-$RELEASE_VERSION"
+        git commit -am "bump version to $RELEASE_VERSION"
+        git push --set-upstream origin "release-$RELEASE_VERSION"
+
+    - name: Release PR
+      run: |
+        gh pr create --title "Release v${RELEASE_VERSION}" --body "Bump version to $RELEASE_VERSION" --base "$TARGET_REF"
+      env:
+        TARGET_REF: ${{ github.event.inputs.base_ref }}
+
+    - name: Dev Cargo.toml
+      run: |
+        sed -i 's/^version = ".*"/version = "${{ env.NEXT_DEV_VERSION }}"/' Cargo.toml
+        cargo metadata >/dev/null
+
+    - name: Dev Commit
+      run: |
+        set -euo pipefail
+        git checkout -b "release-$RELEASE_VERSION-post"
+        git commit -am "bump version to $NEXT_DEV_VERSION"
+        git push --set-upstream origin "release-$RELEASE_VERSION-post"
+
+    - name: Dev PR
+      run: |
+        gh pr create --title "Prepare v${NEXT_DEV_VERSION}" --body "Bump version to $NEXT_DEV_VERSION" --base "release-$RELEASE_VERSION"
```

Comment:
The "Dev *" steps don't quite work: the squash commit of the release version causes them to conflict once they change to target main.

Still, it's good enough for now.